internal/server/evaluator.go

```diff
@@ -434,3 +436,57 @@ func matchesBool(c storage.EvaluationConstraint, v string) (bool, error) {
 
 	return false, nil
 }
+
+func matchesDateTime(c storage.EvaluationConstraint, v string) (bool, error) {
+	switch c.Operator {
+	case flipt.OpNotPresent:
+		return len(strings.TrimSpace(v)) == 0, nil
+	case flipt.OpPresent:
+		return len(strings.TrimSpace(v)) != 0, nil
+	}
+
+	// can't parse an empty string
+	if v == "" {
+		return false, nil
+	}
+
+	d, err := tryParseDateTime(v)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: we should consider parsing this at creation time since it doesn't change and it doesnt make sense to allow invalid constraint values
+	value, err := tryParseDateTime(c.Value)
+	if err != nil {
+		return false, err
+	}
+
+	switch c.Operator {
+	case flipt.OpEQ:
+		return value.Equal(d), nil
+	case flipt.OpNEQ:
+		return !value.Equal(d), nil
+	case flipt.OpLT:
+		return d.Before(value), nil
+	case flipt.OpLTE:
+		return d.Before(value) || value.Equal(d), nil
+	case flipt.OpGT:
+		return d.After(value), nil
+	case flipt.OpGTE:
+		return d.After(value) || value.Equal(d), nil
+	}
+
+	return false, nil
+}
+
+func tryParseDateTime(v string) (time.Time, error) {
+	if d, err := time.Parse(time.RFC3339, v); err == nil {
+		return d, nil
+	}
+
+	if d, err := time.Parse(time.DateOnly, v); err == nil {
```

Comment:
looks like `time.DateOnly` was added in [Go 1.20](https://tip.golang.org/doc/go1.20#minor_library_changes) 😮‍💨

maybe we can drop support for 1.19?

Reply:
Agreed. I cant see any real concern for us having to support more than one version of Go.